packages/components/src/templates/next/components/complex/Accordion/Accordion.tsx

```diff
@@ -1,17 +1,49 @@
-import { BiChevronDown } from "react-icons/bi"
+"use client"
 
-import type { AccordionProps } from "~/interfaces"
+import type { VariantProps } from "tailwind-variants"
+import { useState } from "react"
+import { BiMinus, BiPlus } from "react-icons/bi"
+
+import type { AccordionProps as BaseAccordionProps } from "~/interfaces"
+import { tv } from "~/lib/tv"
 import { Prose } from "../../native"
 
-const Accordion = ({ summary, details }: AccordionProps) => {
+const createAccordionStyles = tv({
+  slots: {
+    details:
+      "group border-y border-divider-medium px-4 py-5 has-[+_details]:border-b-0",
+    summary:
+      "prose-headline-lg-medium flex list-none flex-row items-center justify-between gap-3 text-base-content-strong outline-none outline outline-0 outline-offset-2 outline-brand-interaction hover:cursor-pointer focus-visible:outline-2",
+    icon: "h-6 w-6 flex-shrink-0",
+    content: "pt-5 text-base-content-strong",
+  },
+})
+
+const accordionStyles = createAccordionStyles()
+
+interface AccordionProps
+  extends BaseAccordionProps,
+    VariantProps<typeof createAccordionStyles> {
+  name?: string
+}
+
+const Accordion = ({ summary, details, name }: AccordionProps) => {
+  const [isOpen, setIsOpen] = useState(false)
+  const Icon = isOpen ? BiMinus : BiPlus
+
   return (
-    <details className="border-y border-divider-medium hover:cursor-pointer has-[+_details]:border-b-0 [&_>_summary_>_svg]:open:rotate-180">
-      <summary className="my-[0.125rem] flex list-none flex-row items-center gap-3 px-4 py-5 font-medium text-content outline-none text-paragraph-01">
+    <details
+      name={name}
+      onToggle={(e) => setIsOpen(e.currentTarget.open)}
+      open={isOpen}
```

Comment:
question(non-blocking): are we able to utilise just CSS for this and leave it to the browser to handle the disclosure? Was thinking along the lines of `[&_>_summary_>_#plus]:open:hidden [&_>_summary_>_#minus]:open:block`.

Reply:
good idea. done in [2505bad](https://github.com/opengovsg/isomer/pull/450/commits/2505bad3ab6c83e6e50f2f5bf642863d2532514d)